### PR TITLE
[doc] Add sample CasC configuration for checkmarx-plugin

### PR DESCRIPTION
**Pull request by kleph**
Add a sample CasC configuration for checkmarx-plugin.

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [X ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [X ] Ensure that the pull request title represents the desired changelog entry
- [X ] Please describe what you did
- [ N/A] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [N/A ] Link to relevant pull requests, esp. upstream and downstream changes
- [N/A ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

**Comment by casz** [comment]
https://github.com/jenkinsci/configuration-as-code-plugin/commit/d8f838ad971d93c4e0bc2b3d1bb257c5008c88ac.patch
@kleph 
You might want to set up an email alias on your GitHub account 🥇 
Or look into fixing your git user.email config 😅 
to get proper credit for your commits 🏆 
https://help.github.com/articles/adding-an-email-address-to-your-github-account/
https://help.github.com/articles/setting-your-commit-email-address-in-git
then to amend your commits: https://stackoverflow.com/a/3042512

File: demos/checkmarx/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          description: "login/pwd for checkmarx"
+          username: jenkins_checkmarx
+          password: "StrongPassword"
+
+unclassified:
+  cxscanbuilder:
+    credentialsId: checkmarx_credential
+    serverUrl: https://checkmarx.url.tld
+    enableCertificateValidation: true
+    filterPattern: '!**/_cvs/**/*, !**/.svn/**/*,   !**/.hg/**/*,   !**/.git/**/*,  !**/.bzr/**/*, !**/bin/**/*,
```
Comment (casz) [inline demos/checkmarx/README.md]: please use literal
```suggestion
    filterPattern: -|
    my multiline string
```

File: demos/checkmarx/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# configure checkmarx-plugin
+
```
Comment (v1v) [inline demos/checkmarx/README.md]: Can you add the URL to the plugin with something like the below suggestion?

```suggestion
Basic configuration of the [Checkmarx plugin](https://plugins.jenkins.io/checkmarx)
```

**Comment by v1v** [comment]
https://github.com/jenkinsci/configuration-as-code-plugin/pull/1118 might be a better candidate for this particular demo

File: demos/checkmarx/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+credentials:
+  system:
+    domainCredentials:
+      - credentials:
+        - usernamePassword:
+          scope: GLOBAL
+          id: checkmarx_credential
+          description: "login/pwd for checkmarx"
+          username: jenkins_checkmarx
+          password: "StrongPassword"
```
Comment (timja) [inline demos/checkmarx/README.md]: use secret interpolation not hardcoding
```suggestion
          password: "${CHECKMARX_PASSWORD}"
```

File: demos/checkmarx/README.md (hunk 2 above)
Comment (kleph) [reply demos/checkmarx/README.md]: Sure ! 
done in : 512f236

I just saw that version 8.90.4 is public now. There should be new configuration options to specify a proxy to reach the checkmarx server.
I will update the demo again on monday (from a tested and validated config ;) )

File: demos/checkmarx/README.md (hunk 1 above)
Comment (kleph) [reply demos/checkmarx/README.md]: Thanks @casz for pointing the email aliases issue. I have updated both configurations (local and githib ;) )

I totally forgot about the litteral syntax ! It should be cleaner.
I will push it on monday, after some validation with a test jenkins.

Comment (casz) [reply demos/checkmarx/README.md]: @kleph if you follow @v1v our CI will do the validation for you

File: demos/checkmarx/README.md (hunk 3 above)
Comment (kleph) [reply demos/checkmarx/README.md]: I am not sure to understand your suggestion.
This is the secret definition at system level. It's the only place it appears.

File: integrations/src/test/java/io/jenkins/plugins/casc/CheckmarxTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class CheckmarxTest {
+
+    @Test
+    @ConfiguredWithReadme(value = "checkmarx/README.md")
+    public void configure_artifactory() throws Exception {
```
Comment (timja) [inline integrations/src/test/java/io/jenkins/plugins/casc/CheckmarxTest.java]: Checkmarx?

Comment (kleph) [reply integrations/src/test/java/io/jenkins/plugins/casc/CheckmarxTest.java]: yes, I just fixed it ;-)

File: integrations/src/test/java/io/jenkins/plugins/casc/CheckmarxTest.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class CheckmarxTest {
+
+    @Test
+    @ConfiguredWithReadme(value = "checkmarx/README.md")
```
Comment (v1v) [inline integrations/src/test/java/io/jenkins/plugins/casc/CheckmarxTest.java]: Cosmetic change:

```suggestion
    @ConfiguredWithReadme("checkmarx/README.md")
```

File: demos/checkmarx/README.md (hunk 3 above)
Comment (kleph) [reply demos/checkmarx/README.md]: OK, this should be provided by other means

**Comment by sladyn98** [comment]
@kleph gentle ping, are you still working on this PR, the build is failing due to some dependency error.

**Comment by kleph** [comment]
Hello, @sladyn98, 

I am more an integrator than a developper and those dependencies problem are barely understandable to me. I tried to let IDE do some magic, without success ;-)

I would happyly pass the lead of this PR, if anyone is willing to finish the integration of this sample.

**Comment by jetersen** [comment]
I'll see what I can do @kleph

**Pull request closed without merging**